theano/tensor/tests/test_opt.py

```diff
@@ -3962,23 +3962,33 @@ def setUp(self):
     def test_local_mul_switch_sink(self):
         c = T.dscalar()
         idx = 0
-        for condition in [(T.dmatrix('cond'), self.condm), (T.dvector('cond'), self.condv), (T.dscalar('cond'), self.conds)]:
-            for x in [(T.dmatrix('x'), self.xm), (T.dvector('x'), self.xv), (T.dscalar('x'), self.xs)]:
-                y = T.mul(T.switch(condition[0] > 0, 1. * x[0],
-                    0. * x[0]), T.switch(condition[0] > 0, 1.*x[0], T.log(c)*x[0]))
-                f = theano.function([condition[0], x[0], c]
-                    , [y], mode=self.mode)
+        for condition in [(T.dmatrix('cond'), self.condm),
+                          (T.dvector('cond'), self.condv),
+                          (T.dscalar('cond'), self.conds)]:
+            for x in [(T.dmatrix('x'), self.xm), (T.dvector('x'), self.xv),
+                      (T.dscalar('x'), self.xs)]:
+                y = T.mul(T.switch(condition[0] > 0, 1. * x[0], 0. * x[0]),
+                          T.switch(condition[0] > 0,
+                                   1. * x[0], T.log(c) * x[0]))
+                f = theano.function([condition[0], x[0], c],
+                                    [y], mode=self.mode)
                 if type(condition[1]) is list:
                     for i in range(len(condition[1])):
                         res = f(condition[1][i], x[1], -1)
-                        assert (res == numpy.
-                            asarray(self.resm[idx][i])).sum() == self.resm[idx][i].size
+                        assert (res == numpy.asarray(
+                            self.resm[idx][i])).sum() == self.resm[idx][i].size
                 else:
                     res = f(condition[1], x[1], -1)
                     assert (res == numpy.asarray(self.
                         resm[idx])).sum() == self.resm[idx].size
                 idx += 1
 
+    # This case caused a missed optimization in the past.
+    x = T.dscalar('x')
+    y = T.switch(x < 7, x, T.sqrt(x - 7))
+    f = theano.function([x], T.grad(y, x))
+    assert f(5) == 1, f(5)
```

Comment:
The indentation of that seems off.
Also, you probably need to use `self.mode`, as the tests are failing in FAST_COMPILE.